### PR TITLE
Prints input file to error/verbose output if it exists. Closes #13187

### LINKED ISSUE #13187: Printout full path to input file name for RunApp type tests
## Reason

When a test fails, the TestHarness prints the working directory and the full command line. However, often as a developer you want to open the associated input file to look at and work with the failing test. As it stands, you have to piece that information together from the CWD and the command.

## Design

Add a line to the failed output that prints the full path to any input file.

## Impact

Minor: Usability

### PR DESCRIPTION
**Pull request by socratesgorilla**
Closes #13187

Should be mergeable, one line change.

### CODE REVIEW

**Comment by socratesgorilla** [comment]
It says the precheck is failing because of no correct issue, I have no idea why it is throwing that, as the correct issue is linked here.

File: python/TestHarness/TestHarness.py [hunk 1]
```diff
@@ -446,7 +446,7 @@ def printOutput(self, job, color):
         output = ''
         # Print what ever status the tester has at the time
         if self.options.verbose or (job.isFail() and not self.options.quiet):
-            output = 'Working Directory: ' + job.getTestDir() + '\nRunning command: ' + job.getCommand() + '\n'
+            output = 'Working Directory: ' + job.getTestDir() + '\nRunning command: ' + job.getCommand() + '\n' + ('Input File: ' + job.getTestDir() + job.getInputFile() if job.getInputFile() != None else 'No input file provided.')
```
Comment (milljm) [inline python/TestHarness/TestHarness.py:449]: Is there a way we can use line continuation here, and/or use string replacement techniques?

eg:
```python
'Input File: %s' % (os.path.join(job.getTestDir(), job.getInputFile()))
or
'Input File: {}'.format(os.path.join(job.getTestDir(), job.getInputFile()))
```

Also, I think "Input File: None" would suffice here. As in, we might be able to find a clever way to just return None as a literal string and print that. Thoughts?

**Review by milljm — CHANGES_REQUESTED** [review]
Line continuation, and/or string formatting request.

File: python/TestHarness/TestHarness.py (hunk 1 above)
Comment (milljm) [reply python/TestHarness/TestHarness.py:449]: I was probably the one that originally wrote that line of code :) before I knew about string formatting!

Comment (socratesgorilla) [reply python/TestHarness/TestHarness.py:449]: I'm not sure I understand the point of the os.path calls, they just add runtime and the potential for unintended behavior. All this does is print a string of the input file path, where job.getTestDir() gets the path already through os.path, and job.getInputFile() just gets the name of the file. The rest of your changes are totally cosmetic, I don't particularly care if it says "No input file provided" or "Input file: None", it doesn't really matter, just let me know if you want it

Comment (milljm) [reply python/TestHarness/TestHarness.py:449]: Without the os.path.join (using the system path separator) it constructs the path like this:

```
/Users/me/projects/moose/test/tests/test_harnessgood.i
```

when it should be:
```
/Users/me/projects/moose/test/tests/test_harness/good.i
```

We like to use Python's os.path.join, so we are not hard coding Unix-only path separators (for the eventual day we make this all work on Windows, we don't have to come back and switch a bunch of `/` for `\`).

As for 'Input File: None' vs 'No input file provided', your call. I am good either way. Upon first glance, I figured there was a way to do something like:

```python
output += 'Input File: %s' % (the path || str(job.getInputFile()))
```

**Pull request closed without merging**